sklearn/linear_model/tests/test_logistic.py

```diff
@@ -775,44 +776,54 @@ def test_logistic_regressioncv_class_weights(weight, class_weight, global_random
+        n_samples=300,
+        n_features=5,
+        n_informative=3,
+        n_classes=2,
+        random_state=global_random_seed,
     )
+    rng = np.random.RandomState(global_random_seed)
+    sw = np.ones(y.shape[0])
     sample_weight = y + 1
 
     for LR in [LogisticRegression, LogisticRegressionCV]:
-        kw = {"random_state": 42, "fit_intercept": False}
+        kw_weighted = {"random_state": global_random_seed, "fit_intercept": False}
+        kw_repeated = kw_weighted.copy()
+        sw[:100] = rng.randint(0, 5, size=100)
+        X_repeated = np.repeat(X, sw.astype(int), axis=0)
+        y_repeated = np.repeat(y, sw.astype(int), axis=0)
         if LR is LogisticRegressionCV:
-            kw.update({"Cs": 3, "cv": 3})
-
+            # We weight the first fold 2 times more.
+            groups_weighted = np.r_[np.full(100, 0), np.full(100, 1), np.full(100, 2)]
+            splits_weighted = list(LeaveOneGroupOut().split(X, groups=groups_weighted))
+            kw_weighted.update({"Cs": 3, "cv": splits_weighted})
+
+            groups_repeated = np.repeat(groups_weighted, sw.astype(int), axis=0)
+            splits_repeated = list(
+                LeaveOneGroupOut().split(X_repeated, groups=groups_repeated)
+            )
+            kw_repeated.update({"Cs": 3, "cv": splits_repeated})
         # Test that passing sample_weight as ones is the same as
         # not passing them at all (default None)
-        for solver in ["lbfgs", "liblinear"]:
-            clf_sw_none = LR(solver=solver, **kw)
-            clf_sw_ones = LR(solver=solver, **kw)
-            clf_sw_none.fit(X, y)
-            clf_sw_ones.fit(X, y, sample_weight=np.ones(y.shape[0]))
-            assert_allclose(clf_sw_none.coef_, clf_sw_ones.coef_, rtol=1e-4)
-
-        # Test that sample weights work the same with the lbfgs,
-        # newton-cg, newton-cholesky and 'sag' solvers
-        clf_sw_lbfgs = LR(**kw, tol=1e-5)
-        clf_sw_lbfgs.fit(X, y, sample_weight=sample_weight)
-        for solver in set(SOLVERS) - set(["lbfgs"]):
-            clf_sw = LR(solver=solver, tol=1e-10 if solver == "sag" else 1e-5, **kw)
-            # ignore convergence warning due to small dataset with sag
-            with ignore_warnings():
-                clf_sw.fit(X, y, sample_weight=sample_weight)
-            assert_allclose(clf_sw_lbfgs.coef_, clf_sw.coef_, rtol=1e-4)
+        # Note doesn't work with sag and saga
+        for solver in ["lbfgs", "liblinear", "newton-cg", "newton-cholesky"]:
+            clf_sw_weighted = LR(solver=solver, **kw_weighted)
+            clf_sw_repeated = LR(solver=solver, **kw_repeated)
+            clf_sw_weighted.fit(X, y, sample_weight=sw)
+            clf_sw_repeated.fit(X_repeated, y_repeated)
+            if LR is LogisticRegressionCV:
+                assert_allclose(clf_sw_weighted.scores_[1], clf_sw_repeated.scores_[1])
+            assert_allclose(clf_sw_weighted.coef_, clf_sw_repeated.coef_, rtol=1e-4)
```

Comment:
`rtol=1e-4` seems like a large tolerance level for comparing coefficients...

Maybe the tolerance of the solvers is too high, and the two models do not converge properly enough to be able to test 

I did some local experiments, and I think we could always use:

```python
    max_iter = 10_000
    tol = 1e-12
```

for all estimators in this test and the check for `assert_allclose` on coefs with `rtol=1e-5` and `atol=1e-8`, including with `sag` and `saga` solvers.